api/src/main/java/io/hyperfoil/api/config/BenchmarkBuilder.java

```diff
@@ -180,7 +184,7 @@ private void checkDeadlock(Phase phase, Map<String, Phase> phases) {
       toProcess.add(phase);
       while (!toProcess.isEmpty()) {
          Phase p = toProcess.poll();
-         Stream.concat(p.startAfter.stream(), p.startAfterStrict.stream()).forEach(name -> {
+         Stream.concat(Stream.concat(p.startAfter.stream(), p.startAfterStrict.stream()), p.startWithDelay != null ? Stream.of(p.startWithDelay.phase) : Stream.of()).forEach(name -> {
```

Comment:
please add a comment here as well.
You can check if it is worthy to make the deps list as part of the `Phase` API

Reply:
good idea, I will add `getDependencies()` method in `Phase` class to return all phases' dependencies.